src/AxelarGateway.sol

```diff
@@ -107,16 +111,23 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
         string memory name,
         string memory symbol,
         uint8 decimals,
-        uint256 cap
+        uint256 cap,
+        address tokenAddr
     ) internal {
         require(tokenAddresses(symbol) == address(0), 'TOKEN_EXIST');
 
-        bytes32 salt = keccak256(abi.encodePacked(symbol));
-        address token = address(new BurnableMintableCappedERC20{ salt: salt }(name, symbol, decimals, cap));
+        if (tokenAddr == address(0)) {
+            bytes32 salt = keccak256(abi.encodePacked(symbol));
+            tokenAddr = address(new BurnableMintableCappedERC20{ salt: salt }(name, symbol, decimals, cap));
+        } else {
+            require(!_isTokenDeployed(tokenAddr), 'TOKEN_EXIST');
+            _setTokenExternal(tokenAddr);
+        }
 
-        _setTokenAddress(symbol, token);
+        _setTokenAddress(symbol, tokenAddr);
+        _setTokenDeployed(tokenAddr);
```

Comment:
Shouldn't you do this `_setTokenDeployed(tokenAddr);` in the `if (tokenAddr == address(0))` block? So:
```solidity
        if (tokenAddr == address(0)) {
            _setTokenDeployed(symbol, tokenAddr = address(new BurnableMintableCappedERC20{ salt: keccak256(abi.encodePacked(symbol)) }(name, symbol, decimals, cap)));
        } else {
            require(!_isTokenDeployed(tokenAddr), 'TOKEN_EXIST');
            _setTokenExternal(tokenAddr);
        }
        
        _setTokenAddress(symbol, tokenAddr);
```